### DISCUSSION TITLE
Alexa and HUE-CLIP-PI

### DISCUSSION THREAD

**Discussion by MCMK74** [andesse/hue-clip-api.node-red-flows#37] [Q&A] [answered]
Hello Andreas, 
first of all MANY THANKS for your great job doing here. I love it to find a solution for me where i get more my hands on and being more flexible. Thank you so much!
I hope you can give me (noob) some hints how i can handle my challenge.
I am trying to connect Alexa node. Setup HUE-CLIP-API is done and is working with your example nodes.
When i connect Alex-node to a copy of your examples, for example (On, 100%, 1 Sek) it does not change anything and sticks to the values of the example, they are fixed values in the JSON think.

Alexa sends payload.brightness 60 or payload.powerState "ON". I think i could handle ON into TRUE without JSON (because i do not have a plan) but how can i make it dynamic so it can handle the alexa-node values?

{
    "on": {
        "on": true
    },
    "dimming": {
        "brightness": 100
    },
    "dynamics": {
        "duration": 1000
    },
    "color": {
        "xy": {
            "x": 0.6101,
            "y": 0.3662
        }
    }
}

Long story short, and i am compete new to JSON, how can i handle the payloads of Alex-node dynamically?

Hope i didnt explained it to confusing!

Danke dir!
VG - Marc

**Comment by andesse** [comment]
Hallo Marc,
Ich mach das mal in Englisch so das andere User auch was damit anfangen können. 

For Alexa Commands you need the "on device activity" feature. This one is listening for everything the echo dot heard. It is in an array, i don't know the exact datapoint. You'll figure it out. 

Then you need to switch out all commands that you say, so maybe "Licht Wohnzimmer 50 Prozent" and the switch sends the payload to the action for hue. (You need to make a action for every light level) or using variables to change it. 

Did you get what I mean?
Probably I can make a subflow for this feature request at the weekend, that you can probably use out of the box. 

Gruß 
Andreas

**Reply by MCMK74** [reply]
Hallo Andreas,

deshalb habe ich die Frage ebenfalls auf EN gestellt, für die anderen User. :)

Thank you for your super fast reply. I am not sure if i get you.
Alexa datapoint is working fine and your API too, but i cannot assign the payloads to the api. You know what i mean?
I got all datapoints for example, this is coming from Alexa-node

![image](https://user-images.githubusercontent.com/116357719/197199670-c5a78d27-0ae6-4488-9b96-687b292ac509.png)

but i do not know how to assign them to the API. I switched the datapoints to the API datapints with no effect. Maybe it has to be done in JSON, but here i do not know how.

This is the result which goes into HEADER V2.
![image](https://user-images.githubusercontent.com/116357719/197202600-52298493-d038-4f1d-ad34-75dbb72b3862.png)


Thank you Andreas!

**Reply by MCMK74** [reply]
Hi Andreas,
i found a way, but think it is not elegant:

![image](https://user-images.githubusercontent.com/116357719/197214576-6cdd14c1-7dad-46a6-afd9-e541c2cd8dde.png)

In case i catch and switch some ranges, maybe in 10 steps, will lead in 10 nodes just to control brightness.
In case this is best way at the moment, do you know how i can get status of brightness? ON/OFF i can see in blue node.

Regards, Marc

**Reply by andesse** [reply]
Ich erkläre es auf deutsch 😂

Du musst hinter der Alexa Node on device listening einen Switch setzen und für jeden Sprachbefehl einen separaten Ausgang setzen. 

Beispiel Du sagst "Alexa Licht wohnen 60" dann muss der Switch "enthält" Licht wohnen 60 auf einem Ausgang haben. Diesen verbindest Du dann mit dem Befehl für die HUE api brightness 60. 

Du musst für jeden Wert einen eigenen API Befehl setzen. Oder das über context und function / change node realisieren. 

Ich bin gleich zu Hause, dann mache ich dir mal ein Beispiel. 
Geb mir ne Stunde:)

**Reply by andesse** [reply]
> Hi Andreas,
> 
> i found a way, but think it is not elegant:
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/116357719/197214576-6cdd14c1-7dad-46a6-afd9-e541c2cd8dde.png)
> 
> 
> 
> In case i catch and switch some ranges, maybe in 10 steps, will lead in 10 nodes just to control brightness.
> 
> In case this is best way at the moment, do you know how i can get status of brightness? ON/OFF i can see in blue node.
> 
> 
> 
> Regards, Marc
> 
> 

Ja, genau so musst du es machen, ich versuche mal später einen eleganteren Weg zu programmieren 

Gruß

**Reply by andesse** [reply]
> Hi Andreas, i found a way, but think it is not elegant:
> 
> ![image](https://user-images.githubusercontent.com/116357719/197214576-6cdd14c1-7dad-46a6-afd9-e541c2cd8dde.png)
> 
> In case i catch and switch some ranges, maybe in 10 steps, will lead in 10 nodes just to control brightness. In case this is best way at the moment, do you know how i can get status of brightness? ON/OFF i can see in blue node.
> 
> Regards, Marc

@MCMK74 
The light/group receivers just output a true or false. The raw Data is coming out of the Orange node, Event-Stream. Debug there

**Comment by andesse** [comment, marked as answer]
@MCMK74  Your workaround is already the best solution for Node-RED. Thar was exactly my Idea how it could work. 
The bad thing is that the Alexa API outputs "twenty percent" as a string instead of a number. Thats kinda annoying. Would it be a Number we could figure out something better for dimming values.

I use my Flow actually to connect my Homematic Switches to HUE and from Ipad Shortcut> MQTT > HUE. Mostly triggering Scenes as recommended. I have for example just some special Voice activated Flows like a "Alexa, Bedtime" that turns on the Bed-Lamps nad turns the motion Sensor off. Or "TV on" that activates a dim TV scene for HUE and turns the AVR and TV on.
As well i use the HUE Buttons for other stuff, and use the Evetstream (Light / Group Receivers) for visual feedback at my wall switches, activating a LED.

Otherwise i just use my Echo-Dot for ON / OFF. I recommend using all Voice features from Alexa to HUE,  i think the best way is to use the HUE Plugin for Alexa instead of doing it in Node-RED.

Hope that helped.

EDIT: Wait, what Alexa node returned the dimming value as a number? 

Gruß und schönes WE
Andreas

**Comment by MCMK74** [comment]
Hi Andreas,
I already implement KNX and other things in node and have had a working scenario with Alexa, Homekit and KNX. 
Now just HUE not working any longer because of bad HUEMAGIC. 
I thought there is maybe a solution to solve it with some variables instead of fixed values. 
Thank you for incredible fast answers! Great!

Yes, alexa returns numbers in brightness! See my Screenshot above. 

Danke dir und schönes WE!
Marc

**Reply by andesse** [reply]
> Hi Andreas, I already implement KNX and other things in node and have had a working scenario with Alexa, Homekit and KNX. Now just HUE not working any longer because of bad HUEMAGIC. I thought there is maybe a solution to solve it with some variables instead of fixed values. Thank you for incredible fast answers! Great!
> 
> Yes, alexa returns numbers in brightness! See my Screenshot above.
> 
> Danke dir und schönes WE! Marc

Hey @MCMK74 

i am pretty sure you can use functions to convert the incoming Alexa message to a value stored in context and used in the action change node. I am just not a programmer and just know basic function coding. A lot of the function parts in this contrib came from @FredBlo . Fred. Maybe you can help? Can you make a function that converts a string „fifty percent“ from a specific datapoint from Alexa to number 50? That would be very helpful and we had a new feature for the contrib

cheers
Andreas

**Reply by andesse** [reply]
@MCMK74 
Ach ja, ganz vergessen, tut mir leid… VIELEN DANK FÜR DAS LOB! :)
Ich habe das Flow Beispiel genau aus dem Grund gemacht den Du genannt hast. 

HueMagic, was im Kern eine super contrib ist, wird einfach nicht gewartet. Das ist wirklich sehr schade. Auch wenn es wieder funktionieren sollte, werde ich nicht zurück gehen. Ich werde Änderungen sofort anpassen, ich sehe die sofort, ich habe zwei Bridges und eine ist auf Beta. 

Der direkte http Request wie ihn die API vorgibt ist um einiges schneller, auch wenn es nicht so komfortabel ist. Mach Dir mal einen Account bei HUE Developer. Die API V2 ist quasi noch Beta, da wird noch einiges kommen und die Doku dort ist echt sehr gut. Wenn die Frage aufkommt wofür Du entwickelst, schreib einfach Node-RED. ;)

Gruß 
Andreas
